From bf32ddf1f26952583d634a8db3aaf802d662d423 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 19 Jul 2024 15:47:07 +0200
Subject: [PATCH 1063/1695] dts: stm32: default enable TAMP and RTC for every
 SoC

Default probe TAMP on every stm32mp1/2 based product as the
configuration of the backup registers is always needed. Also, default
probe RTC to solve TAMP's dependency on it.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2beafd0d7aaa53bf1bd9cf9d1136f33da4891525 dts: stm32: default enable TAMP and RTC for every SoC]
Change-Id: Ia116d16aed52c9dfd8da19ee6fa4cdf9d50df13a
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/394365
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp131.dtsi | 2 --
 core/arch/arm/dts/stm32mp151.dtsi | 1 -
 core/arch/arm/dts/stm32mp211.dtsi | 2 --
 core/arch/arm/dts/stm32mp231.dtsi | 2 --
 core/arch/arm/dts/stm32mp251.dtsi | 2 --
 5 files changed, 9 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index f6d69e86f..0060549ba 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -372,7 +372,6 @@
 			clocks = <&rcc RTCAPB>, <&rcc RTC>;
 			clock-names = "pclk", "rtc_ck";
 			interrupts-extended = <&exti 19 IRQ_TYPE_LEVEL_HIGH>;
-			status = "disabled";
 		};
 
 		bsec: efuse@5c005000 {
@@ -435,7 +434,6 @@
 			clocks = <&rcc RTCAPB>;
 			ranges;
 			st,backup-zones = <10 5 17>;
-			status = "disabled";
 
 			nvram: nvram@5c00a100 {
 				compatible = "st,stm32mp15-tamp-nvram";
diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index fccc90d57..cb69651da 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -365,7 +365,6 @@
 			reg = <0x5c004000 0x400>;
 			clocks = <&rcc RTCAPB>, <&rcc RTC>;
 			clock-names = "pclk", "rtc_ck";
-			status = "disabled";
 		};
 
 		bsec: efuse@5c005000 {
diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 730c65e74..e5a1c98da 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -844,7 +844,6 @@
 			clocks = <&rcc CK_BUS_RTC>, <&rcc RTC_CK>;
 			clock-names = "pclk", "rtc_ck";
 			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
-			status = "disabled";
 		};
 
 		tamp: tamp@46010000 {
@@ -856,7 +855,6 @@
 			#size-cells = <1>;
 			ranges;
 			st,backup-zones = <24 24 24 24 12 12 8>;
-			status = "disabled";
 
 			nvram: nvram@46010100 {
 				compatible = "st,stm32mp25-tamp-nvram";
diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 0e500a2bd..dd1f0cfad 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -1001,7 +1001,6 @@
 			clocks = <&rcc CK_BUS_RTC>, <&rcc RTC_CK>;
 			clock-names = "pclk", "rtc_ck";
 			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
-			status = "disabled";
 		};
 
 		tamp: tamp@46010000 {
@@ -1013,7 +1012,6 @@
 			#size-cells = <1>;
 			ranges;
 			st,backup-zones = <24 24 24 24 12 12 8>;
-			status = "disabled";
 
 			nvram: nvram@46010100 {
 				compatible = "st,stm32mp25-tamp-nvram";
diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 038d059c4..083115749 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -1124,7 +1124,6 @@
 			clocks = <&rcc CK_BUS_RTC>, <&rcc RTC_CK>;
 			clock-names = "pclk", "rtc_ck";
 			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
-			status = "disabled";
 		};
 
 		tamp: tamp@46010000 {
@@ -1136,7 +1135,6 @@
 			#size-cells = <1>;
 			ranges;
 			st,backup-zones = <24 24 24 24 12 12 8>;
-			status = "disabled";
 
 			nvram: nvram@46010100 {
 				compatible = "st,stm32mp25-tamp-nvram";
-- 
2.47.2

